### ISSUE TITLE
flatMapPrefix doesn't propagate attributes to the nested flow

### ISSUE THREAD

**Issue by eyalfa** [akka/akka#30374]
the following tests demonstrates the issue:

```
test("flatMapPrefix attributes") {
    case object CustomAttribute extends Attribute

    class WithAttr[A] extends GraphStage[FlowShape[A, (A, Option[CustomAttribute.type])]]{
      override val shape: FlowShape[A, (A, Option[CustomAttribute.type])] = FlowShape(Inlet[A]("in"), Outlet[(A, Option[CustomAttribute.type])]("out"))
      override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) with InHandler with OutHandler {
        val attr = inheritedAttributes.get[CustomAttribute.type]

        setHandlers(shape.in, shape.out, this)

        override def onPush(): Unit = push(shape.out, grab(shape.in) -> attr)

        override def onPull(): Unit = pull(shape.in)
      }
    }
    def withAttr[A] = Flow fromGraph new WithAttr[A]

    implicit val mat = Materializer(implicitActorSystem)

    Source
      .single("1")
      .via(withAttr)
      .map(_._2)
      .withAttributes(Attributes(CustomAttribute))
      .runWith(Sink.head)
      .futureValue should be (Some(CustomAttribute))

    Source
      .single("1")
      .flatMapPrefix(0){_ =>
        Flow[String]
          .via(withAttr)
          .map(_._2)
      }
      .withAttributes(Attributes(CustomAttribute))
      .runWith(Sink.head)
      .futureValue should be (Some(CustomAttribute))
  }
```

the second assertion fails because the nested flow is materialized only with the default attributes.

seems like `FlattenMerge` implements this correctly:
```
val graph = Source.fromGraph(source).to(sinkIn.sink)
interpreter.subFusingMaterializer.materialize(graph, defaultAttributes = enclosingAttributes)
```

so I suggest implementing the same behaviour for `flatMapPrefix`

**Comment by eyalfa** [comment]
@johanandren , @jrudolph , 
you were the reviewers of this feature, can you please comment on this?

**Comment by jrudolph** [comment]
Have you found any mention of this feature in the documentation? I think it's a non-obvious feature but probably one that someone found useful if it has been implemented like this for FlattenMerge.

**Comment by eyalfa** [comment]
@jrudolph , I actually introduced the feature 😎 
just found out this behaviour is missing, looked at other similar operators and found what I believe is(/should be) the standard.

will u be willing to review a PR for this?

**Comment by johanandren** [comment]
I think it makes sense to align with other operators that materialize nested graphs. Please go ahead and PR 👍

**Comment by eyalfa** [comment]
hmm, this seems to be recently fixed by @armanbilge in #30161 .
I can still contribute tests for the `flatMapPrefix` operator, it you're willing to accept them.
